src/LaunchDarkly.Client/LdClient.cs

```diff
@@ -258,8 +258,8 @@ protected virtual void Dispose(bool disposing)
         {
             Logger.LogInformation("Closing LaunchDarkly client.");
             //We do not have native resource, so the boolean parameter can be ignored.
-            if (_eventStore is EventProcessor)
-                ((_eventStore) as IDisposable).Dispose();
+
+            _eventStore.Dispose();
```

Comment:
Giving all event stores the chance to cleanup gracefully.

Reply:
I had written that originally, but it felt wrong.  It contradicts most of the .NET framework, so I rewrote it before sending you the PR.  Here are 5 examples in the .NET framework:

HttpClient - https://msdn.microsoft.com/en-us/library/hh875108(v=vs.118).aspx
NetworkStream - https://msdn.microsoft.com/en-us/library/te7e60bx(v=vs.110).aspx
StreamReader - https://msdn.microsoft.com/en-us/library/gg712952(v=vs.110).aspx
StreamWriter - https://msdn.microsoft.com/en-us/library/gg712853(v=vs.110).aspx
XmlWriterSettings.CloseOutput - https://msdn.microsoft.com/en-us/library/system.xml.xmlwritersettings.closeoutput(v=vs.110).aspx

The general pattern is dispose child disposables by default, unless otherwise explicitly told not to.

I get that it's a breaking change, but a worth wild one as it promotes good practices.